### PR TITLE
Change stock images to proper symbolic icons

### PR DESCRIPTION
**Pull request by GabMus**

### CODE REVIEW

**Comment by xou816** [comment]
Hi! Thank you for your contribution, I didn't know where to find the proper symbolic icons, this is great! :)
I broke master (...oops), I'll fix it later today and notify you so you can rebase, but I am fairly confident it will build just fine :)

**Comment by xou816** [comment]
(there are a couple other UI files that could benefit from this change, now that I think  of it)

**Comment by GabMus** [comment]
> Hi! Thank you for your contribution

You're welcome!

> I didn't know where to find the proper symbolic icons

I'm not magical either, I use [this awesome app](https://gitlab.gnome.org/World/design/icon-library/)!

> I broke master (...oops), I'll fix it later today and notify you so you can rebase, but I am fairly confident it will build just fine :)

Hopefully, please tell me when I can rebase

File: src/app/components/song/song.ui [hunk 1]
```diff
@@ -63,13 +63,7 @@
...
         <property name="relief">none</property>
         <property name="always-show-image">True</property>
-        <child>
-          <object class="GtkImage">
-            <property name="visible">True</property>
-            <property name="can-focus">False</property>
-            <property name="stock">gtk-media-play</property>
-          </object>
-        </child>
+        <property name="icon-name">media-playback-start-symbolic</property>
```
Comment (xou816) [inline src/app/components/song/song.ui:66]: I think we need to keep the GtkImage child:
`Gtk-ERROR **: 21:33:04.295: failed to add UI: .:156:53 Invalid property: GtkButton.icon-name`

File: src/window.ui [hunk 2]
```diff
@@ -153,13 +153,7 @@
...
                         <property name="valign">center</property>
                         <property name="relief">none</property>
-                        <child>
-                          <object class="GtkImage">
-                            <property name="visible">True</property>
-                            <property name="can-focus">False</property>
-                            <property name="stock">gtk-media-previous</property>
-                          </object>
-                        </child>
+                        <property name="icon-name">media-skip-backward-symbolic</property>
```
Comment (xou816) [inline src/window.ui:156]: (same as above, GtkImage needed)

File: src/window.ui [hunk 3]
```diff
@@ -207,13 +201,7 @@
...
                         <property name="valign">center</property>
                         <property name="relief">none</property>
-                        <child>
-                          <object class="GtkImage">
-                            <property name="visible">True</property>
-                            <property name="can-focus">False</property>
-                            <property name="stock">gtk-media-next</property>
-                          </object>
-                        </child>
+                        <property name="icon-name">media-skip-forward-symbolic</property>
```
Comment (xou816) [inline src/window.ui:204]: (same as above, GtkImage needed)

File: src/window.ui [hunk 4]
```diff
@@ -327,13 +315,7 @@ audio-volume-medium-symbolic</property>
...
                 <property name="halign">start</property>
                 <property name="valign">center</property>
-                <child>
-                  <object class="GtkImage">
-                    <property name="visible">True</property>
-                    <property name="can-focus">False</property>
-                    <property name="stock">gtk-go-back</property>
-                  </object>
-                </child>
+                <property name="icon-name">go-previous-symbolic</property>
```
Comment (xou816) [inline src/window.ui:318]: (same as above, GtkImage needed)

**Comment by xou816** [comment]
> I'm not magical either, I use [this awesome app](https://gitlab.gnome.org/World/design/icon-library/)!

I had never heard of that tool, it's really cool! (even has code snippets, nice!)

> Hopefully, please tell me when I can rebase

...finally fixed it, phew! You can rebase :)

File: src/app/components/details/details.ui [hunk 5]
```diff
@@ -38,7 +38,9 @@
                     <property name="height-request">100</property>
                     <property name="visible">True</property>
                     <property name="can-focus">False</property>
-                    <property name="stock">gtk-missing-image</property>
+                    <!-- I'm taking the freedom to use a music symbolic icon
```
Comment (xou816) [inline src/app/components/details/details.ui:41]: perfect, I like it better that way!

File: src/app/components/song/song.ui (hunk 1 above)
Comment (GabMus) [reply src/app/components/song/song.ui]: Ah, sorry, that's for GTK4, fixed

**Comment by GabMus** [comment]
Fixed the missing images and rebased, let me know how it looks now

**Comment by xou816** [comment]
Almost there :) It's missing the visible property (and Glade also adds can-focus, but I'm not sure it is truly needed):
```xml
<object class="GtkImage">
<property name="visible">True</property>
<property name="can-focus">False</property>
<property name="icon-name">media-skip-forward-symbolic</property>
</object>
```

But it's definitely looking better :)

**Comment by GabMus** [comment]
You're right, visible is missing and should be there, but can-focus should default to false anyway, right? Plus, word of advice: stop using glade, it's not really supported and you'll have to clean up the mess glade made to your ui files once you'll start porting to gtk4

**Comment by GabMus** [comment]
Done, hopefully everything is in place now!

**Review by xou816 — APPROVED** [review]

**Comment by xou816** [comment]
Right, I read about Glade being sub optimal but... I don't know if I lack experience with Gtk or if these claims are exaggerated, but Glade has never caused me any trouble so far, quite the opposite :) I guess I'll see when I get to work on porting to Gtk4!

Anyway, many thanks for your work, let's merge this!

**Pull request merged**